pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
```

Comment:
Yep, perfect (the fspath stuff happens inside `_stringify_path`). Let's hope that works.